javascript/javascript1/week3/cactus_io.js

```diff
@@ -0,0 +1,128 @@
+const smartPhoneActivities = [];
+const usageLimit = 120;
+
+function addActivity(activity, duration, activities) {
+    const errorMessages = [];
+    const errorMessage = [
+        "ACTIVITY MUST BE A STRING!",
+        "DURATION MUST BE A NUMBER!",
+    ]
+
+    if (typeof activity !== "string") {
+        errorMessages.push(errorMessage[0]);
+    } 
+    
+    if (typeof duration !== "number") {
+        errorMessages.push(errorMessage[1]);
+    } 
+    
+    if (errorMessages.length > 0) {
+        console.log(errorMessages);
+        return;
+    } else {
+        activities.push({
```

Comment:
To avoid confusion, for now you can just use the "global" variable `smartPhoneActivities ` here:
```suggestion
        smartPhoneActivities.push({
```

Reply:
Regarding this, I used to call the declared variable (in this example, the array), but then I don't see the point in adding parameters to the functions I create, because all of them work on the outside variable. Because of this, I thought I would make more sense to work on the parameter inside the function and then pass the variable as an argument when calling the function. I know in this case it doesn't make any differences, but for me it makes things more meaningful!